chainstate/src/rpc/mod.rs

```diff
@@ -40,99 +40,126 @@ use serialization::hex_encoded::HexEncoded;
     async fn get_utxo(&self, outpoint: UtxoOutPoint) -> RpcResult<Option<TxOutput>>;
 
-    /// Submit a block to be included in the chain
+    /// Submit a block to be included in the blockchain.
+    ///
+    /// Note that the submission does not circumvent any validation process.
+    /// This function is used by the wallet to submit valid blocks after successful staking.
     #[method(name = "submit_block")]
     async fn submit_block(&self, block_hex: HexEncoded<Block>) -> RpcResult<()>;
 
     /// Invalidate the specified block and its descendants.
+    ///
+    /// Use this function with caution, as invalidating a block that the network approves
+    /// of can lead to staying behind.
     #[method(name = "invalidate_block")]
     async fn invalidate_block(&self, id: Id<Block>) -> RpcResult<()>;
 
     /// Reset failure flags for the specified block and its descendants.
     #[method(name = "reset_block_failure_flags")]
     async fn reset_block_failure_flags(&self, id: Id<Block>) -> RpcResult<()>;
 
-    /// Get block height in main chain
+    /// Get block height in mainchain, given a block id.
     #[method(name = "block_height_in_main_chain")]
     async fn block_height_in_main_chain(
         &self,
         block_id: Id<GenBlock>,
     ) -> RpcResult<Option<BlockHeight>>;
 
-    /// Get best block height in main chain
+    /// Get best block height in mainchain.
     #[method(name = "best_block_height")]
     async fn best_block_height(&self) -> RpcResult<BlockHeight>;
 
     /// Returns last common block id and height of two chains.
-    /// Returns None if no block indexes are found and therefore the last common ancestor is unknown.
+    /// Returns None if no blocks are found and therefore the last common ancestor is unknown.
     #[method(name = "last_common_ancestor_by_id")]
     async fn last_common_ancestor_by_id(
         &self,
         first_block: Id<GenBlock>,
         second_block: Id<GenBlock>,
     ) -> RpcResult<Option<(Id<GenBlock>, BlockHeight)>>;
 
+    /// Returns the balance of the pool associated with the given pool id.
+    ///
+    /// The balance contains both delegated balance and staker balance.
+    /// Returns `None` (null) if the pool is not found.
     #[method(name = "stake_pool_balance")]
     async fn stake_pool_balance(&self, pool_id: PoolId) -> RpcResult<Option<Amount>>;
 
+    /// Returns the balance of the staker (pool owner) of the pool associated with the given pool address.
+    ///
+    /// This excludes the delegation balances.
+    /// Returns `None` (null) if the pool is not found.
     #[method(name = "staker_balance")]
     async fn staker_balance(&self, pool_id: PoolId) -> RpcResult<Option<Amount>>;
 
+    /// Given a pool defined by a pool address, and a delegation address,
+    /// returns the amount of coins owned by that delegation in that pool.
     #[method(name = "delegation_share")]
     async fn delegation_share(
         &self,
         pool_id: PoolId,
```

Comment:
hmm, actually pool_id is redundant as delegation id is unique across pools. Maybe we should remove `get_stake_pool_delegation_share` from ChainstateInterface and use `get_stake_delegation_balance` everywhere

Reply:
I wouldn't worry about this now. Though feel free to open a PR for this.